### PR TITLE
Add Comment to each self-approved merge request

### PR DESCRIPTION
**Pull request by matejsmisek**

### CODE REVIEW

File: server/src/NotifyIfMergeRequestIsSelfApproved.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+				};
+			},
+		);
+
+		const message = `
+#postmergeapprove
+====
+Merge request was self-approved by @${
+			mergeRequest.author.username
+		}. You need to get second approve from anybody but you. They should reply l thread with 'Approved' message to post-approve this MR.
```
Comment (fprochazka) [inline server/src/NotifyIfMergeRequestIsSelfApproved.ts:105]: `They should reply to this message in thread`

File: server/src/NotifyIfMergeRequestIsSelfApproved.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+Code owners:
+${codeowners
```
Comment (fprochazka) [inline server/src/NotifyIfMergeRequestIsSelfApproved.ts:107]: there should be another newline to separate markdown paragraphs and ideally, each rule should be prefixed with `*` so it's nicely formatted and readable, right now it looks like this 

image

File: server/src/NotifyIfMergeRequestIsSelfApproved.ts (hunk 1 above)
Comment (matejsmisek) [reply server/src/NotifyIfMergeRequestIsSelfApproved.ts]: Fixed

**Review by fprochazka — APPROVED** [review]

**Review by fprochazka — APPROVED** [review]

File: server/src/GitlabApi.ts [hunk 3]
```diff
@@ -119,6 +120,31 @@ export interface MergeRequestInfo extends MergeRequest {
...
 	approvals_required: number;
 	approvals_left: number;
+	approved_by: MergeRequestUserApprover[];
+}
+
+export interface MergeRequestUserApprover {
+	user: {
+		username: string;
+		id: number;
+	};
```
Comment (janedbal) [inline server/src/GitlabApi.ts:130]: I see 3 shapes with username and id, arent those actually the same shapes in the API? If so, I'd use single interface for it here.

File: server/src/NotifyIfMergeRequestIsSelfApproved.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+		mergeRequest.project_id,
+		mergeRequest.iid,
+	);
+
+	// If there are multiple approvals we don't check if author is among them. Not marking it as self-approved
+	if (approvals.approved_by.length > 1) {
+		return false;
+	}
+
+	for (const approval of approvals.approved_by) {
```
Comment (janedbal) [inline server/src/NotifyIfMergeRequestIsSelfApproved.ts:24]: A little confusing to iterate over 0|1 results, but ok

**Review by janedbal — APPROVED** [review]

**Pull request merged**